cpu/fe310/cpu.c

```diff
@@ -17,379 +17,65 @@
-    printf("stack start: 0x%08x\n", (unsigned int)(sched_active_thread->stack_start));
-    printf("stack end  : 0x%08x\n", (unsigned int)(sched_active_thread->stack_start + sched_active_thread->stack_size));
-#endif
-
-    printf("  address:      data:\n");
-
-    do {
-        printf("  0x%08x:   0x%08x\n", (unsigned int) sp, (unsigned int) *sp);
-        sp++;
-        count++;
-    } while (*sp != STACK_MARKER);
-
-    printf("current stack size: %i words\n", count);
-}
-
-int thread_isr_stack_usage(void)
-{
-    return 0;
-}
-
-void *thread_isr_stack_pointer(void)
-{
-    return NULL;
-}
-
-void *thread_isr_stack_start(void)
-{
-    return NULL;
-}
-
-/**
- * @brief Call context switching at thread exit
- *
- * This is called is two situations: 1) after the initial main and idle threads
- * have been created and 2) when a thread exits.
- *
- */
-void cpu_switch_context_exit(void)
-{
-    /* enable interrupts */
-    irq_enable();
-
-    /* force a context switch to another thread */
-    thread_yield_higher();
-    UNREACHABLE();
-}
-
-void thread_yield_higher(void)
-{
-    /* Use SW intr to schedule context switch */
-    CLINT_REG(CLINT_MSIP) = 1;
-
-    /* Latency of SW intr can be 4-7 cycles; wait for the SW intr */
-    __asm__ volatile ("wfi");
-}
-
-/**
- * @brief Print heap statistics
- */
-void heap_stats(void)
-{
-    extern char _heap_start; /* defined in linker script */
-    extern char _heap_end;   /* defined in linker script */
 
-    long int heap_size = &_heap_end - &_heap_start;
-    struct mallinfo minfo = mallinfo();
-    printf("heap: %ld (used %u, free %ld) [bytes]\n",
-           heap_size, minfo.uordblks, heap_size - minfo.uordblks);
+    /* Disable SPI flash mode */
+    SPI0_REG(SPI_REG_FCTRL) &= ~SPI_FCTRL_EN;
+
+    /* Enable QPI mode by sending command to flash */
+    SPI0_REG(SPI_REG_TXFIFO) = 0x35;
+
+    /* begin{code-style-ignore} */
+    SPI0_REG(SPI_REG_FFMT) =               /* setup "Fast Read Quad I/O (QPI mode)"         */
+        SPI_INSN_CMD_EN         |          /* Enable memory-mapped flash                    */
+        SPI_INSN_ADDR_LEN(3)    |          /* 25LP03D read commands have 3 address bytes    */
+        SPI_INSN_PAD_CNT(6)     |          /* 25LP03D Table 6.11 Read Dummy Cycles = 6      */
+        SPI_INSN_CMD_PROTO(SPI_PROTO_Q) |  /* 25LP03D Table 8.1 "Instruction                */
+        SPI_INSN_ADDR_PROTO(SPI_PROTO_Q) | /*  Set" shows mode for cmd, addr, and           */
+        SPI_INSN_DATA_PROTO(SPI_PROTO_Q) | /*  data protocol for given instruction          */
+        SPI_INSN_CMD_CODE(0xEB) |          /* Set the instruction to "Fast Read Quad I/O"   */
+        SPI_INSN_PAD_CODE(0x00);           /* Dummy cycle sends 0 value bits                */
+    /* end{code-style-ignore} */
+
+    /* Re-enable SPI flash mode */
+    SPI0_REG(SPI_REG_FCTRL) |= SPI_FCTRL_EN;
+
+    /* Adjust the SPI clk divider for to boost flash speed */
+  //  SPI0_REG(SPI_REG_SCKDIV) = SCKDIV;
```

Comment:
Did you try boosting the SPI clock speed with the Quad mode?  The default SCKDIV is pretty slow